### ISSUE TITLE
SC2165/SC2167 ignored loop index flagged as overridden

### ISSUE THREAD

**Issue by bboozzoo** [koalaman/shellcheck#1847]
Shellcheck raises SC2167, SC2165 on a script which I believe is ok.

- shellcheck version v0.7.0+git126.7b99823 (snap edge channel)
- [x] The rule's wiki page does not already cover this
- [x] I tried on shellcheck.net and verified that this is still a problem on the latest commit

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/sh

for _ in $(seq 1 2); do
    # ignoring index
    for _ in $(seq 1 5); do
        # ignoring index again
        echo "foo"
    done
done
```

#### Here's what shellcheck currently says:
```

In - line 3:
for _ in $(seq 1 2); do
^-^ SC2167: This parent loop has its index variable overridden.


In - line 5:
    for _ in $(seq 1 5); do
    ^-^ SC2165: This nested loop overrides the index variable of its parent.
```

Otherwise thanks for building this great tool!

**Comment by matthewpersico** [comment]
Doesn’t matter if you use the var’s value or not. It’s still being overridden. Stick an `echo “$_”` in each loop and watch what happens.

**Comment by bboozzoo** [comment]
> Doesn’t matter if you use the var’s value or not. It’s still being overridden. Stick an `echo “$_”` in each loop and watch what happens.

That's true, though AFAICT it's not entirely inconsistent between bash/dash/zsh. What I'm missing is consistency with how `_` is treated in this snipped:
```
for _ in $(seq 1 2) ; do
    echo "foo"
done
```
As I understand, by convention `_` is treated as ignored, and shellcheck follows this, otherwise it'd be flagged due to `SC2034`. In the nested snippet `_` isn't ignored as I'd expect it to be.
